src/plugindata.h

```diff
@@ -263,8 +265,66 @@ void plugin_configure_single(GtkWidget *parent);
 void plugin_help(void);
 void plugin_cleanup(void);
 
+/** Called by Geany when a plugin library is loaded.
+ *
+ * This is the original entry point. Implement and export this function to be loadable at all.
+ * Then fill in GeanyPlugin::info and GeanyPlugin::funcs of the passed @p plugin. Finally
+ * GEANY_PLUGIN_REGISTER() and specify a minimum supported API version.
+ *
+ * For all glory details please read @ref howto.
+ *
+ * @param plugin The unique plugin handle to your plugin. You must set some fields here.
+ * @param module The GModule associated with your plugin library.
+ * @param geany_api_version The API version exposed by the Geany instance that loads your plugin.
+ *           This probably not the same as the one your plugin was compiled against.
```

Comment:
> Yes, there could be a separate function, but the param works just as well and comes for free.

Actually it probably works less well, as if someone uses it it's likely to be outside of `geany_load_module()`, so having the param here means its value has to be forwarded, while a function will be available globally.

Maybe we'd need to know exactly how we expect people to use it to provide the best API for it.

> I expect it to be used more and more going forward.

Well, I for one will probably *never* use it.  The only use case for it it supporting older Geany versions, which while nice also means conditional code, which generally leads to complexity.  Even more so with dynamic conditionals, that are trickier to even check.

BTW, what would we actually access dynamically?  Accessing functions is really not easy, as there is no real way to discover them if they weren't available at build time -- and if a plugin was using them, it then depends on that symbol anyway.  This would work for dynamic stuff like signals, but I can't really seem to think of anything more right now -- but maybe for accessing unofficial API as it might change without changing ABI, though also without changing API.